gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/language/grammar/ArgumentVisitor.java

```diff
@@ -100,6 +101,31 @@ public GValue<String> parseString(final GremlinParser.StringNullableArgumentCont
         }
     }
 
+    /**
+     * Alternative to visitIntegerArgument which promotes types to either Long or GValue<Long>
```

Comment:
I read this comment as: `visitIntegerArgument` will do type promotion but this method won't. Not sure if this is confusing for others as well.

Reply:
That's the opposite of my intent, I rewrote the comment to be less ambiguous.